### PR TITLE
Collapsable component

### PR DESCRIPTION
**Pull request by mbmjertan**
# Description

Adds a React component `Collapsable`. 

It's like a simpler version of `CollapsableComponentUseToggle`, without support for use toggles etc., and allows for a simple way to group options in accordion-like components.

# Screenshots / Videos

https://user-images.githubusercontent.com/1742806/157243316-ad78c1f9-3c1c-4b06-b690-77aa42f19c95.mov

# Linked documentation PR

See story.js, README for usage examples.

### CODE REVIEW

File: scripts/components/collapsable/collapsable.scss [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+	&.is-open {
+		box-shadow: 0 0 0 1px var(--wp-admin-theme-color, #194BD4);
+		padding: 0.5rem;
+
+		margin-left: -0.5rem;
+		margin-right: -0.5rem;
+
+    .es-collapsable__trigger {
```
Comment (goranalkovic-infinum) [inline scripts/components/collapsable/collapsable.scss:34]: Why is this separate from the `&__trigger` above?

**Review by goranalkovic-infinum — CHANGES_REQUESTED** [review]
Left 1 question, nice job!

File: scripts/components/collapsable/collapsable.scss (hunk 1 above)
Comment (mbmjertan) [reply scripts/components/collapsable/collapsable.scss:34]: This is taken from `CollapsableComponentUseToggle` styles, and the reasoning behind this is that these styles should only get applied to this selector: `.es-collapsable.is-open .es-collapsable__trigger`.

I can write it down using `$this`. Not sure if any alternatives come to your mind.
```scss
&.is-open {
  // ...
  #{$this}__trigger { 
    // ... 
  }
}
```

Comment (goranalkovic-infinum) [reply scripts/components/collapsable/collapsable.scss:34]: Ah nevermind, the indents confused me, all good! 😄

**Review by goranalkovic-infinum — DISMISSED** [review]

**Review by iobrado — APPROVED** [review]

**Review by dingo-d — APPROVED** [review]

**Review by dadadavorin — APPROVED** [review]

**Review by gabriel-glo — APPROVED** [review]

**Pull request merged**